modules/xmpp/strophe.jingle.js

```diff
@@ -22,6 +22,144 @@ const logger = getLogger(__filename);
+ * @param {*} jsonMessageXml The XML node for the "json-message" element.
+ * @returns nothing.
+ */
+function expandSourcesFromJson(iq, jsonMessageXml) {
+
+    let json;
+    try {
+        json = JSON.parse(jsonMessageXml.textContent);
+    } catch (error) {
+        logger.error(`json-message XML contained invalid JSON, ignoring: ${jsonMessageXml.textContent}`);
+        return;
+    }
+
+    if (!json || !json.sources) {
+        // It might be a message of a different type, no need to log.
+        return;
+    }
+
+    // This is where we'll add "source" and "ssrc-group" elements. Create them elements if they don't exist.
+    let videoRtpDescription = getOrCreateRtpDescription(iq, 'video');
+    let audioRtpDescription = getOrCreateRtpDescription(iq, 'audio');
+
+    for (let owner in json.sources) {
+        let ownerSources = json.sources[owner];
+        // The video sources, video ssrc-groups, audio sources and audio ssrc-groups are encoded in that order in the
+        // elements of the array.
+        let videoSources = ownerSources && ownerSources.length > 0 && ownerSources[0];
+        let videoSsrcGroups = ownerSources && ownerSources.length > 1 && ownerSources[1];
+        let audioSources = ownerSources && ownerSources.length > 2 && ownerSources[2];
+        let audioSsrcGroups = ownerSources && ownerSources.length > 3 && ownerSources[3];
+
+        if (videoSources && videoSources.length > 0) {
+            for (let i = 0; i < videoSources.length; i++) {
+                videoRtpDescription.appendChild(createSourceExtension(owner, videoSources[i]))
+            }
+        }
+        if (videoSsrcGroups && videoSsrcGroups.length > 0) {
+            for (let i = 0; i < videoSsrcGroups.length; i++) {
+                videoRtpDescription.appendChild(createSsrcGroupExtension(videoSsrcGroups[i]))
+            }
+        }
+        if (audioSources && audioSources.length > 0) {
+            for (let i = 0; i < audioSources.length; i++) {
+                audioRtpDescription.appendChild(createSourceExtension(owner, audioSources[i]))
+            }
+        }
+        if (audioSsrcGroups && audioSsrcGroups.length > 0) {
+            for (let i = 0; i < audioSsrcGroups.length; i++) {
+                audioRtpDescription.appendChild(createSsrcGroupExtension(audioSsrcGroups[i]))
+            }
+        }
+    }
+}
+
+/**
+ * Creates a "source" XML element for the source described in compact JSON format in [sourceCompactJson].
+ * @param {*} owner the endpoint ID of the owner of the source.
+ * @param {*} sourceCompactJson the compact JSON representation of the source.
+ * @returns the created "source" XML element.
+ */
+function createSourceExtension(owner, sourceCompactJson) {
+    let node = $build('source', {xmlns: 'urn:xmpp:jingle:apps:rtp:ssma:0', ssrc: sourceCompactJson.s})
+    if (sourceCompactJson.m) {
+        node.c('parameter', {name: 'msid', value: sourceCompactJson.m}).up()
+    }
+    if (sourceCompactJson.c) {
```

Comment:
I think we should be ok with skipping this. WDYT ?
`cname` is not used anywhere, once the ssrc is added to remote desc and sRD is done, the track event fired by the browser uses the `msid` as the stream id which is then used for creating/identifying the remote track.

Reply:
I'd be more than happy to remove it everywhere :)

Reply:
I'd rather do it in a separate PR though -- starting with Jicofo.

Reply:
Will you still need to make changes in Jicofo if the client stops sending it ? That way we don't have to worry about mobile catching up with latest LJM version.

Reply:
No, if it is not used by the client then the changes can be done independently.